aws/data_source_aws_route_tables_test.go

```diff
@@ -0,0 +1,119 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+)
+
+func TestAccDataSourceAwsRouteTables(t *testing.T) {
+	rInt := acctest.RandIntRange(0, 256)
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckVpcDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceAwsRouteTablesConfig(rInt),
+			},
+			{
+				Config: testAccDataSourceAwsRouteTablesConfigWithDataSource(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("data.aws_route_tables.selected", "ids.#", "3"),
```

Comment:
This check is currently failing:

```
=== RUN   TestAccDataSourceAwsRouteTables
--- FAIL: TestAccDataSourceAwsRouteTables (11.13s)
    testing.go:518: Step 1 error: Check failed: Check 1/2 error: data.aws_route_tables.selected: Attribute 'ids.#' expected "3", got "4"
```

If I had to venture a guess, its because VPCs include a default route table in addition to the 3 being created in the test.

Reply:
@bflad of course you are correct.  I've updated the test cases to account for that, now it succeeds in my tests.
```
jmpb:aws jaukeman$ TF_ACC=1 go test -v -run TestAccDataSourceAwsRouteTables
=== RUN   TestAccDataSourceAwsRouteTables
--- PASS: TestAccDataSourceAwsRouteTables (54.93s)
PASS
ok      github.com/terraform-providers/terraform-provider-aws/aws       54.974s
```